ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/BonsaiInMemoryWorldState.java

```diff
@@ -150,7 +150,12 @@ public void persist(final BlockHeader blockHeader) {
     final Hash newWorldStateRootHash = rootHash(localUpdater);
     archive
         .getTrieLogManager()
-        .saveTrieLog(archive, localUpdater, newWorldStateRootHash, blockHeader, this);
+        .saveTrieLog(
+            archive,
+            localUpdater,
+            newWorldStateRootHash,
+            blockHeader,
+            (BonsaiPersistedWorldState) this.copy());
```

Comment:
is this copy can impact the performance ? I see some synchronized method , call to DB . if @ahamlat  can confirm that it is safe

Reply:
Agreed, an alternate implementation is to directly subscribe and unsubscribe to the underlying worldstate storage.  This is cleaner code-wise, but if there is a performance impact we can directly sub/unsub to the storage itself.